### PR TITLE
Issue #146: SuppressionPatchXpathFilter: ArrayTrailingComma's context strategy

### PR DESCRIPTION
**Pull request by HuGanghui**
Issue #146: SuppressionPatchXpathFilter: ArrayTrailingComma's context strategy

common pattern https://github.com/checkstyle/patch-filters/issues/8#issuecomment-665565892

```
                 2  // context violation
-                ,};
+                };
 
         int[] a3 = new int[]{
                 1,
                 2  // context violation
-                ,
         };

```
Problem: `};` and ` ,` are not child node of 2

### CODE REVIEW

File: src/test/java/com/puppycrawl/tools/checkstyle/filters/SuppressionPatchXpathFilterTest.java [hunk 1]
```diff
@@ -149,4 +149,15 @@ public void testOuterTypeNumber() throws Exception {
         testByConfig(
                 "sizes/OuterTypeNumber/context/defaultContextConfig.xml");
     }
+
+    @Test
+    @Ignore("until https://github.com/checkstyle/patch-filters/issues/146")
```
Comment (timurt) [inline src/test/java/com/puppycrawl/tools/checkstyle/filters/SuppressionPatchXpathFilterTest.java:154]: Why it is ignored?

File: src/test/resources/com/puppycrawl/tools/checkstyle/filters/suppressionpatchxpathfilter/ArrayTrailingComma/context/defaultContext.patch [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+         int[] a2 = new int[]{
+                 1,
+                 2  // context violation
+-                ,};
++                };
+ 
+         int[] a3 = new int[]{
+                 1,
+                 2  // context violation
+-                ,
```
Comment (HuGanghui) [inline src/test/resources/com/puppycrawl/tools/checkstyle/filters/suppressionpatchxpathfilter/ArrayTrailingComma/context/defaultContext.patch:15]: Problem: only deletion line, can be solved in #162

File: src/test/java/com/puppycrawl/tools/checkstyle/filters/SuppressionPatchXpathFilterTest.java (hunk 1 above)
Comment (HuGanghui) [reply src/test/java/com/puppycrawl/tools/checkstyle/filters/SuppressionPatchXpathFilterTest.java:176]: https://github.com/checkstyle/patch-filters/pull/147#issue-456147762

File: src/test/resources/com/puppycrawl/tools/checkstyle/filters/suppressionpatchxpathfilter/ArrayTrailingComma/context/defaultContextConfig.xml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+<module name="Checker">
+    <module name="TreeWalker">
+        <module name="ArrayTrailingComma">
+            <property name="alwaysDemandTrailingComma" value="true"/>
+        </module>
+
+        <module name="com.puppycrawl.tools.checkstyle.filters.SuppressionPatchXpathFilter">
+            <property name="file" value="${tp}/defaultContext.patch" />
+            <property name="strategy" value="context" />
+            <property name="neverSuppressedChecks" value="ArrayTrailingCommaCheck" />
```
Comment (rdiachenko) [inline src/test/resources/com/puppycrawl/tools/checkstyle/filters/suppressionpatchxpathfilter/ArrayTrailingComma/context/defaultContextConfig.xml:14]: Can we use `checkNameForContextStrategyByTokenOrParent` instead?

Comment (HuGanghui) [reply src/test/resources/com/puppycrawl/tools/checkstyle/filters/suppressionpatchxpathfilter/ArrayTrailingComma/context/defaultContextConfig.xml:14]: no, but we can use ancestor node, described here https://github.com/checkstyle/patch-filters/issues/302#issuecomment-674113880

**Pull request merged**